src/algokit/cli/deploy.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import os
+from pathlib import Path
+
+import click
+
+from algokit.core.deploy import DEPLOY_CONFIGS, execute_deploy_command, load_deploy_command, load_deploy_config
+
+logger = logging.getLogger(__name__)
+
+
+class HiddenSecret:
+    def __init__(self, secret: str = ""):
+        self.secret = secret
+
+    def __str__(self) -> str:
+        return "*" * len(self.secret)
+
+
+DEPLOYER_KEY = "DEPLOYER_MNEMONIC"
+DISPENSER_KEY = "DISPENSER_MNEMONIC"
+
+
+@click.command("deploy")
+@click.argument("network", type=click.Choice(list(DEPLOY_CONFIGS)), default="localnet", required=True)
+@click.option(
+    "--deployer_mnemonic",
+    required=True,
+    default=lambda: HiddenSecret(os.environ.get(DEPLOYER_KEY, "")),
```

Comment:
can you use click support for env-vars here? https://click.palletsprojects.com/en/8.1.x/options/#values-from-environment-variables

Reply:
I've refactored it but using a different approach, please check the latest code @achidlow